src/Main/UserContext.js

```diff
@@ -1,15 +1,126 @@
-import React, { useState } from "react";
-
+import React, { useEffect, useState } from "react";
+import firebase from "../components/Firebase/firebase";
+import data from "../Data/survey";
 const UserContext = React.createContext();
 
 function UserProvider(props) {
-  const user = useState({ name: "nouser", email: "" }),
-    wrapperClass = useState("flex-row"),
-    menuPlacement = useState("menu-right");
-  const surveyChoices = useState({Transport: "Car", Food: "Omnivore"})
+  const user = useState({
+    name: undefined,
+    email: undefined,
+    password: undefined,
+  });
+  const loggedIn = useState(false);
+  const [surveyChoicesData, setSurveyChoices] = useState([
+    4,
+    4,
+    4,
+    4,
+    4,
+    4,
+    4,
+    4,
+    4,
+    4,
+    4,
+    4,
+  ]);
+  const pledges = useState({
+    transport: [],
+    food: [],
+    goods: [],
+    household: [],
+  });
+
+  const footprint = useState();
+
+  /* This is a massively shitty way of doing it, but it was the best way 
+  I could think of that was also sort of readable, without having to 
+  redo the survey format due to changes in Data/survey.js.
+  After having fought with syncing survey values and checked status to surveyChoices,
+  I don't wanna go down that dark hall again just yet */
+
+  function calcFootprint(theme) {
+    //console.log(theme)
+    let surveyResult = 0;
+    let index = 0;
+    switch (theme) {
+      case "transport":
+        index = 0; //redundant, for reading purposes
+        break;
+      case "food":
+        index = 3;
+        break;
+      case "good":
+        index = 6;
```

Comment:
can be rewritten with a dictionary, no? {"transport":0, "food":3, "good":6, etc...} . or?

Reply:
By dictionary you mean object?

Reply:
Sure.

    const indexReference = {
      "transport": 0,
      "food": 3,
      "good": 6,
      "household": 9
    }
    let surveyResult = 0;
    let index = indexReference[theme];
    
    I personally don't find it more readable than a switch, but as you wish.